stl/inc/vector

```diff
@@ -3718,6 +3718,155 @@ _NODISCARD _CONSTEXPR20 _Iter_diff_t<_VbIt> _Count_vbool(_VbIt _First, const _Vb
+
+    const bool _IsSingleBlockSource = _VbFirst == _VbLast;
+    const bool _IsSingleBlockDest   = _VbDest == _DestEnd._Myptr;
+    const bool _IsRightShift        = _Dest._Myoff < _First._Myoff;
+    if (_IsSingleBlockSource) {
+        // We already excluded _First == _Last, so here _Last._Myoff > 0 and the shift is safe
+        const auto _LastSourceMask = static_cast<_Vbase>(-1) >> (_VBITS - _Last._Myoff);
+        const auto _SourceMask     = _FirstSourceMask & _LastSourceMask;
+        const auto _SourceShift    = _IsRightShift ? _First._Myoff - _Dest._Myoff : _Dest._Myoff - _First._Myoff;
+        const auto _SourceVal      = _IsRightShift ? (*_VbFirst & _SourceMask) >> _SourceShift //
+                                                   : (*_VbFirst & _SourceMask) << _SourceShift;
+        if (_IsSingleBlockDest) {
+            const auto _DestMask = _FirstDestMask | _LastDestMask;
+            *_VbDest             = (*_VbDest & _DestMask) | _SourceVal;
+        } else {
+            *_VbDest = (*_VbDest & _FirstDestMask) | _SourceVal;
+            ++_VbDest;
+
+            const auto _LastShift     = _Last._Myoff - _DestEnd._Myoff;
+            const auto _LastSourceVal = (*_VbFirst & _SourceMask) >> _LastShift;
+            *_VbDest                  = (*_VbDest & _LastDestMask) | _LastSourceVal;
+        }
+
+        return _DestEnd;
+    } else if (_IsSingleBlockDest) {
+        const auto _SourceShift = _IsRightShift ? _First._Myoff - _Dest._Myoff : _Dest._Myoff - _First._Myoff;
+        const auto _SourceVal   = _IsRightShift ? (*_VbFirst & _FirstSourceMask) >> _SourceShift //
+                                                : (*_VbFirst & _FirstSourceMask) << _SourceShift;
+
+        const auto _DestMask = _FirstDestMask | _LastDestMask;
+        if (_Last._Myoff != 0) {
+            const auto _LastSourceMask = static_cast<_Vbase>(-1) >> (_VBITS - _Last._Myoff);
+            const auto _LastShift      = _DestEnd._Myoff - _Last._Myoff;
+            const auto _LastSourceVal  = (*_VbLast & _LastSourceMask) << _LastShift;
+            *_VbDest                   = (*_VbDest & _DestMask) | _SourceVal | _LastSourceVal;
+        } else {
+            *_VbDest = (*_VbDest & _DestMask) | _SourceVal;
+        }
+
+        return _DestEnd;
+    }
+
+    const auto _UnalignedFirstBits = _First._Myoff & _Vbase{7};
+    const auto _UnalignedDestBits  = _Dest._Myoff & _Vbase{7};
+    if (_UnalignedFirstBits == _UnalignedDestBits) { // _First and _Dest have matching char alignment, so use memmove
+        const auto _UnalignedLastBits = _Last._Myoff & _Vbase{7};
+
+        // What is the Strict Aliasing Rule and Why do we care?
+        char* _VbFirst_ch = reinterpret_cast<char*>(_VbFirst) + (_First._Myoff - _UnalignedFirstBits) / 8;
+        char* _VbDest_ch  = reinterpret_cast<char*>(_VbDest) + (_Dest._Myoff - _UnalignedFirstBits) / 8;
+        char* _VbLast_ch  = reinterpret_cast<char*>(_VbLast) + (_Last._Myoff - _UnalignedLastBits) / 8;

```

Comment:
`reinterpret_cast` and uses of the results are constexpr-incompatible, while this function is intentionally callable in constant evaluation since C++20. Perhaps we need an `if (_STD is_constant_evaluated())` branch.